packages/common/directives/src/filter/filter.directive.ts

```diff
@@ -1,37 +1,54 @@
-import { Directive, ElementRef, HostListener, Input, OnDestroy, OnInit } from '@angular/core';
-import { ControlValueInterceptor, ControlValueInterceptorDescriptor } from '@angular-ru/common/forms';
+import { Directive, ElementRef, HostListener, Input } from '@angular/core';
+import { ControlValueInterceptor } from '@angular-ru/common/forms';
 import { filter, FilterPredicateFn } from '@angular-ru/common/string';
 
 @Directive({
     selector: '[filter]',
     providers: [ControlValueInterceptor]
 })
-export class FilterDirective implements OnInit, OnDestroy {
+export class FilterDirective {
     @Input() public filter: string[] | FilterPredicateFn | RegExp = [];
```

Comment:
could be create alias type

`export type FilterPredicate = string[] | FilterPredicateFn | RegExp`;

Reply:
good idea